### ISSUE TITLE
question: how to undefine Promise global

### ISSUE THREAD

**Issue by zaggino** [eslint/eslint#1921]
I'd like to use `let Promise = require('bluebird');` in `node` env but I keep getting
```
'Promise' is read only                            no-undef
```
How to allow a global to be overridden?
Or even better, revert a env global so I get errors when someone tries to use the native Promise?

**Comment by nzakas** [comment]
You can add this:

```js
/*globals Promise:true*/
```

The true indicates it can be overridden.

Or specify in your config file. See http://eslint.org/docs/configuring/#specifying-globals

**Comment by zaggino** [comment]
It's possible to undefine it when it has been already defined by `node` env?

**Comment by lo1tuma** [comment]
Duplicate of #892

**Comment by malandrew** [comment]
@nzakas How do I allow Promise to be overwritten without also allowing it as a global? You're workaround has the side effect of allowing it as a global, which I don't want.

Can we add this to the `ecmaFeatures` options somehow? The same goes for anything that isn't a reserved keyword like `Symbol` as well.

**Comment by nzakas** [comment]
Promise is always a global, you can't get around that. If you're not using it as a global, then you won't get a warning from this rule.
